dashboard/app/models/sections/section.rb

```diff
@@ -228,7 +228,11 @@ def update_student_sharing(sharing_disabled)
   def name_safe_students
     name_splitter_proc = ->(student) {FullNameSplitter.split(student.name)}
 
-    SafeNames.get_safe_names(students, name_splitter_proc).map do |safe_name_and_student|
+    students_only = students.filter do |user|
+      user.user_type == "student"
```

Comment:
This is a really good callout. I really like the lens of Allowlist vs Blocklist for consistency with how we are doing these user_type checks.

We actually only have two user_types in our [User model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/user.rb#L165), `teacher` and `student`. A `teacher` can be an admin. That being said, I think staying consistent is important here, and since we are effectively "blocking" teachers, this filter should probably filter on `user.user_type != "teacher"`